components/ratgdo/ratgdo.cpp

```diff
@@ -462,9 +462,13 @@ namespace ratgdo {
 
         if (this->obstruction_sensor_detected_) {
             this->door_action(DoorAction::CLOSE);
-        } else if (*this->door_state == DoorState::OPEN) {
-            ESP_LOGD(TAG, "No obstruction sensors detected. Close using TOGGLE.");
-            this->door_action(DoorAction::TOGGLE);
+        } else {
+            if (*this->door_state == DoorState::OPEN || *this->door_state == DoorState::STOPPED) {
+                ESP_LOGD(TAG, "No obstruction sensors detected. Close using TOGGLE.");
+                this->door_action(DoorAction::TOGGLE);
+            } else {
+                this->door_action(DoorAction::CLOSE);
+            }
```

Comment:
It looks like this logic could be combined a bit and it doesn't need to be nested